src/cloudsc_gpu/cloudsc_driver_gpu_scc_field_mod.F90

```diff
@@ -222,28 +232,33 @@ SUBROUTINE CLOUDSC_DRIVER_GPU_SCC_FIELD( &
     CALL TIMER%THREAD_END(TID)
 
     IF(USE_PACKED)THEN
-       CALL FIELD_STATE%DATA_RWONLY%SYNC_HOST_RDWR()
+       CALL FIELD_STATE%AUX%SYNC_HOST()
```

Comment:
The pullback-to-host should mirror the copy-to-device logic; we should call `SYNC_HOST_RDWR` on the FieldGang if `USE_PACKED` is true, and then simply call `SYNC_HOST_RDWR` on all the named children. Once `SYNC_HOST_RDWR` is called on the FieldGang, calling it on the children doesn't trigger any data movement. The same is also true of calling it twice on the same field (`GET_DEVICE_RDONLY` followed by `SYNC_HOST_RDWR` will also not move any data back to host).

I believe this change should make the `SYNC_HOST` methods redundant, so you could then remove those.

Reply:
I don't really understand what difference is between SYNC_HOST and SYNC_HOST_RDWR for the FieldGang. Is it exactly the same as for normal fields? Might have to update this then.

Reply:
`SYNC_HOST` is not a F-API call but rather a type-bound method of the new derived-types you have introduced. I think a `SYNC_HOST` method exists for the corresponding types in ecphys. It makes a lot of sense when adding the F-API calls by hand but if the F-API boilerplate is to be loki generated then we shouldn't add data movement methods to IFS derived types and just call `SYNC_HOST_RDWR/RDONLY` on all the constituent fields.

Calling `SYNC/GET_HOST/DEVICE_*` methods on the FieldGang is functionality equivalent to calling the same methods one-by-one on all of the children, except that we will now do one contiguous copy rather than many strided copies.